gen/definitions/deploy_template.yaml

```diff
@@ -0,0 +1,98 @@
+---
+name: Deploy Template
+rest_endpoint: /dna/intent/api/v2/template-programmer/template/deploy
+no_delete: true
+no_update: true
+no_read: true
+doc_category: Templates
+test_tags: [TEMPLATES]
+attributes:
+  - model_name: templateId
+    type: String
+    id: true
+    description: Main template UUID of versioned template
+    test_value: catalystcenter_template_version.example.id
+    example: 12345678-1234-1234-1234-123456789012
+  - model_name: forcePushTemplate
+    type: Bool
+    description: Force Push Template
+    example: false
+  - model_name: isComposite
+    type: Bool
+    description: Composite template flag
+    example: false
+  - model_name: mainTemplateId
+    type: String
+    description: Main template UUID of versioned template
+    example: 12345678-1234-1234-1234-123456789012
+    exclude_test: true
+  - model_name: memberTemplateDeploymentInfo
+    type: String
```

Comment:
According to the documentation this is an array<undefined>. Should this be a StringList? Do you know what this is being used for?

Reply:
looks like API documentation is wrong, i mean its a array but array of objects like this: 

```
deployment_info_cisco = {
        "targetInfo": [
            {
                "type": "MANAGED_DEVICE_IP",
                "id": "172.27.52.8",
            }
        ],
        "templateId": "c82c301b-4705-4221-9877-ab42d6c6cfea",
        "isComposite": True,
        "memberTemplateDeploymentInfo": [
            {
                "mainTemplateId": "1ef9fc6d-2bf2-4aaf-bd79-7999073f5030",
                "isComposite": False,
                "forcePushTemplate": True,
                "templateId": "0b83bef2-fe49-4641-b382-9a5bc4090b62",
                "targetInfo": [
                    {
                        "type": "MANAGED_DEVICE_IP",
                        "params": {
                            "radiusdata": {
                                "cisco_probe_pwd": "probe_pwd",
                                "shared_key": "sharedkey_pwd",
                                "group_name": "ISE",
                                "servers": [
                                    {"name": "server1", "ip": "1.1.1.1"},
                                    {"name": "server3", "ip": "3.3.3.3"},
                                ],
                            },
                        },
                        "id": "172.27.52.8",
                    }
                ],
            },
            {
                "mainTemplateId": "ea9b2311-b27e-4c27-97cb-f6ce4cf2e1a7",
                "isComposite": False,
                "forcePushTemplate": True,
                "templateId": "bcfccd5f-4508-4f1e-b594-aa0ff473ba38",
                "targetInfo": [
                    {
                        "type": "MANAGED_DEVICE_IP",
                        "params": {
                            "vlans": [
                                {"id": 666, "name": "User_Vlan_666"},
                                {"id": 667, "name": "User_Vlan_667"},
                            ],
                        },
                        "id": "172.27.52.8",
                    }
                ],
            },
        ],
        "forcePushTemplate": True,
    }
```
Do you think we should add that to definition ? This would be a valid payload, but its not coming from documenation but from rather doing reverse engineering on payload which will work

Reply:
Added attributes in definition file to support this payload format